middleware/2faMessageContent.js

```diff
@@ -2,40 +2,71 @@ const hex = require('hexer');
 const nearAPI = require('near-api-js');
 const escapeHtml = require('escape-html');
 
+const TRUNCATE_ARGUMENTS_LENGTH = 247;
+
 const fmtNear = (amount) => nearAPI.utils.format.formatNearAmount(amount, 4) + 'Ⓝ';
 
-const formatArgs = (args) => {
+const formatArgs = (args, isForSmsDelivery = false) => {
+    let output = '';
+
     const argsBuffer = Buffer.from(args, 'base64');
     try {
         const jsonString = argsBuffer.toString('utf-8');
-        const json = JSON.parse(jsonString);
-        if (json.amount) json.amount = fmtNear(json.amount);
-        if (json.deposit) json.deposit = fmtNear(json.deposit);
-        return JSON.stringify(json);
-    } catch(e) {
+        const parsed = JSON.parse(jsonString);
+
+        // Composing new obj in specific order so `amount` and `deposit` will likely be in the first 250chars
+        const { amount, deposit, ...json } = parsed;
+
+        const formattedNear = {
+            ...Object.hasOwnProperty.call(parsed, 'amount') && { amount: fmtNear(amount) },
```

Comment:
@MaximusHaximus not gonna block on this as logic is the same as before, looks like added here:
https://github.com/near/near-contract-helper/pull/244/files

Looks like it's not correct logic however, as there is no guarantee that `deposit` and `amount` fields contain NEAR tokens. Every fungible token has `amount` field and they can have arbitrary number of decimals:
https://explorer.mainnet.near.org/transactions/8MSMkTkHKy77426bTghQX9kRvwb9Xp9ssa3QzCcWggvJ